modules/blocks/README.md

```diff
@@ -0,0 +1,166 @@
+
+Take a step back and consider the ideal workflow for adding a Gravatar. After inserting a new "Gravatar" block, as a user I'd expect it to be shown in some empty state, with an option to add an email address in a [text input](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input). Upon entering an email address, a preview of the image should be shown next to the input. At this point, you might realize that while you'd want some controls to be shown when editing content, the markup included in the published post might not appear the same (your visitors should not see an input field when reading your content). This leads to the first requirement of describing a block: __you will need to provide implementations both for what's to be shown in an editor and what's to be saved with the published content__. You needn't worry about redundant effort here, as concepts of [Elements](../elements/README.md) and componentization provide an avenue for sharing common behaviors.
+
+Now that we've considered user interaction, you should think about the underlying values that determine the markup generated by your block. In our example, the output is affected only when the email address changes. Put another way: __the output of a block is a function of its attributes__. The email address, a simple string, is the only thing we require to be able to generate the image we want to include in the published content. We call these underlying values of a block instance its _attributes_.
+
+With these concepts in mind, let's explore an implementation of our Gravatar block:
+
+```php
+<?php
+// plugin.php
+
+function gravatar_block_enqueue_scripts() {
+	wp_register_script( 'blueimp-md5', 'https://cdnjs.cloudflare.com/ajax/libs/blueimp-md5/2.7.0/js/md5.min.js' );
+	wp_enqueue_script( 'gravatar-block', plugins_url( 'block.js', __FILE__ ), array( 'wp-blocks', 'wp-element', 'blueimp-md5' ) );
+}
+add_action( 'block_enqueue_scripts', 'gravatar_block_enqueue_scripts' );
+```
+
+```js
+// block.js
+
+var el = wp.element.createElement;
+
+function GravatarImage( props ) {
+	var src = md5( props.email );
+
+	return el( 'img', { src: props.src } );
+}
+
+wp.blocks.registerBlock( 'myplugin/gravatar', {
+	edit: function( block ) {
+		var email = block.attributes.email,
+			children;
+
+		function onSubmit( event ) {
+			var input = event.target.querySelector( 'input' );
+			block.setAttributes( { email: input.value } );
+			event.preventDefault();
+		}
+
+		function onBlur( event ) {
+			block.setAttributes( { email: event.target.value } );
+		}
+
+		children = el( 'input', { value: email, onBlur: onBlur } );
+		if ( email ) {
+			children.unshift( GravatarImage( { email: email } ) );
+		}
+
+		return el( 'form', { onSubmit: setEmail }, children );
+	},
+	save: function( block ) {
+		return GravatarImage( { email: block.attributes.email } );
```

Comment:
How would you imagine us getting the HTML from this? we `render` it to a dom node and user `innerHTML`? or maybe we'd use `renderToString`?